source/common/router/router.cc

```diff
@@ -1525,6 +1529,10 @@ void Filter::doRetry() {
     downstream_headers_->setEnvoyAttemptCount(attempt_count_);
   }
 
+  if (config_.add_request_date_header_) {
```

Comment:
add TODO to move to extensions when pluggable retry options are available

Reply:
Good call, updated!